### PR TITLE
Add sync pre-commit config

### PR DESCRIPTION
**Pull request by isamu-takagi**
## PR Type

Improvement

## Description

Modify workflow.

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [x] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s).
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: .github/sync-files.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@
     - source: .clang-format
     - source: .markdown-link-check.json
     - source: .markdownlint.yaml
+    - source: .pre-commit-config.yaml
```
Comment (kenji-miyake) [inline .github/sync-files.yaml:13]: Since autowarefoudation/autoware doesn't have source code, there is no clang-formar or cpplint, is that okay?

Comment (isamu-takagi) [reply .github/sync-files.yaml:13]: Which repository should I sync with? Is it better to copy other files from autoware_common?

Comment (kenji-miyake) [reply .github/sync-files.yaml:13]: In this case, I believe from autoware_common is better.

Comment (isamu-takagi) [reply .github/sync-files.yaml:13]: Thank you. Modified it. https://github.com/tier4/tier4_ad_api_adaptor/pull/15/commits/df64d5da8c823e79a2721aa199c714c51092179f

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**